RadarSDK/RadarVerifiedLocationToken.m

```diff
@@ -0,0 +1,95 @@
+                                events:(NSArray<RadarEvent *> *_Nonnull)events
+                                 token:(NSString *_Nonnull)token
+                             expiresAt:(NSDate *_Nonnull)expiresAt
+                             expiresIn:(NSTimeInterval)expiresIn
+                                passed:(BOOL)passed {
+    self = [super init];
+    if (self) {
+        _user = user;
+        _events = events;
+        _token = token;
+        _expiresAt = expiresAt;
+        _expiresIn = expiresIn;
+        _passed = passed;
+    }
+    return self;
+}
+
+- (instancetype _Nullable)initWithObject:(id _Nonnull)object {
+    if (!object || ![object isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+    
+    NSDictionary *dict = (NSDictionary *)object;
+    
+    RadarUser *user;
+    NSArray<RadarEvent *> *events;
+    NSString *token;
+    NSDate *expiresAt;
+    NSTimeInterval expiresIn = 0;
+    BOOL passed = NO;
+    
+    id userObj = dict[@"user"];
+    if (userObj && [userObj isKindOfClass:[NSDictionary class]]) {
+        user = [[RadarUser alloc] initWithObject:userObj];
+        
+        NSDictionary *userDict = (NSDictionary *)userObj;
+        id actualUpdatedAtObj = userDict[@"actualUpdatedAt"];
+        if (actualUpdatedAtObj && [actualUpdatedAtObj isKindOfClass:[NSString class]]) {
+            NSString *actualUpdatedAtStr = (NSString *)actualUpdatedAtObj;
+            NSDate *actualUpdatedAt = [RadarUtils.isoDateFormatter dateFromString:actualUpdatedAtStr];
+            expiresIn = [expiresAt timeIntervalSinceDate:actualUpdatedAt];
+        }
+        
+        passed = user && user.fraud && user.fraud.passed && user.country && user.country.passed && user.state && user.state.passed;
+    }
+    
+    id eventsObj = dict[@"events"];
+    if (eventsObj && [eventsObj isKindOfClass:[NSArray class]]) {
+        events = [RadarEvent eventsFromObject:eventsObj];
+    }
+    
+    id tokenObj = dict[@"token"];
+    if (tokenObj && [tokenObj isKindOfClass:[NSString class]]) {
+        token = (NSString *)tokenObj;
+    }
+    
+    id expiresAtObj = dict[@"expiresAt"];
+    if (expiresAtObj && [expiresAtObj isKindOfClass:[NSString class]]) {
+        NSString *expiresAtStr = (NSString *)expiresAtObj;
+        expiresAt = [RadarUtils.isoDateFormatter dateFromString:expiresAtStr];
+    }
+    
+    if (user && events && token && expiresAt) {
+        return [[RadarVerifiedLocationToken alloc] initWithUser:user events:events token:token expiresAt:expiresAt expiresIn:expiresIn passed:passed];
+    }
+    
+    return nil;
+}
+
+- (NSDictionary *)dictionaryValue {
+    NSMutableDictionary *dict = [NSMutableDictionary new];
+    dict[@"user"] = [self.user dictionaryValue];
+    dict[@"events"] = [RadarEvent arrayForEvents:self.events];
+    dict[@"token"] = self.token;
+    dict[@"expiresAt"] = [RadarUtils.isoDateFormatter stringFromDate:self.expiresAt];
```

Comment:
Do we want a null check as `stringFromDate` expects a non-nullable `NSDate`?

Also nit, but I slightly prefer:
`[[RadarUtils isoDateFormatter] stringFromDate:self.expiresAt];`

as it matches our more standard objective-c style.

Reply:
Could go either way here. We are using dot notation for `self.expiresAt` in the same line